lib/view_component/system_test_helpers.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module ViewComponent
+  module SystemTestHelpers
+    include TestHelpers
+    include Capybara::DSL
+
+    def page
+      Capybara.current_session
+    end
+
+    def visit_rendered_component_in_browser(component, **options)
+      layout = options[:layout] || false
+
+      opts = {
+        layout: layout,
+        locals: {
+          render_args: {
+            component: component,
+            template: "view_components/preview",
+            hide_preview_source: true
+          }
+        }
+      }
+      html = controller.render_to_string("view_components/preview", opts)
+
+      # Add 'tmp/' directory if it doesn't exist to store the rendered component html
+      Dir.mkdir("tmp/") unless Dir.exist?("tmp/")
+
+      # Write to temporary file to contain fully rendered component
+      # within a browser
+      file = Tempfile.new(["rendered_#{component.class.name}", ".html"], "tmp/")
```

Comment:
Thinking about alternatives, is it possible to use some kind of static asset middleware to serve HTML files that we write to a specific directory? e.g. `./tmp/view_component_integrations`

I think that would help us mitigate some security issues and maybe allow us to remove some complexity.

Reply:
I'am not entirely sure about the asset middleware but I believe we can use a specific directory :).

I'll make adjustments to use a specific directory. But could you elaborate a bit more on the static assets middleware? I'am afraid I'am not too versed in making middleware.